src/chunk_processors.py

```diff
@@ -181,3 +196,151 @@ def __call__(self, chunks: Sequence[Chunk]) -> Sequence[Chunk]:
             chunk.heading = current_subheading or current_heading
 
         return new_chunks
+
+
+class RemoveRegexPattern(PipelineComponent):
+    """
+    Remove text from chunks that matches a regex pattern.
+
+    If the chunk's entire text matches the regex pattern, remove the chunk. If it
+    just contains text with the regex pattern, replace the pattern with the text
+    specified.
+    """
+
+    def __init__(self, pattern: str, replace_with: str) -> None:
+        self.pattern = pattern
+        self.replace_with = replace_with
+
+    def __call__(self, chunks: list[Chunk]) -> list[Chunk]:
+        """Run regex pattern removal."""
+        if not hasattr(self, "pattern"):
+            raise ValueError("No pattern was set. Please set a pattern in __init__.")
+
+        new_chunks: list[Chunk] = []
+
+        for chunk in chunks:
+            # If the entire text matches the pattern, skip this chunk
+            if re.match(f"^{self.pattern}$", chunk.text):
+                continue
+
+            # Otherwise remove any matches of the pattern from the text
+            new_text = re.sub(self.pattern, self.replace_with, chunk.text)
+            new_chunk = chunk.model_copy()
+            new_chunk.text = new_text.strip()
+
+            # Match cases where the text is made up of multiple repeated instances of
+            # the pattern.
+            if new_chunk.text:
+                new_chunks.append(new_chunk)
+
+        return new_chunks
+
+
+class RemoveFalseCheckboxes(RemoveRegexPattern):
+    """
+    Remove false checkboxes from the Azure output.
+
+    These are :selected: and :unselected: patterns.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(pattern=r"\s?:(?:un)?selected:\s?", replace_with=" ")
+
+
+class CombineSuccessiveSameTypeChunks(PipelineComponent):
+    """Combines successive chunks of the same type in a sequence of chunks."""
+
+    def __init__(self, chunk_types: list[str], text_separator="\n") -> None:
```

Comment:
suggestion: when i read this for the first time i had to read the code to figure out whether `chunk_types` was a list of chunk types to combine, or to keep, or to discard, or to ignore! a tiny bit more specificity in the parameter names could make this component much easier for other folks to pick up and reuse :)